### ISSUE TITLE
Feature Perturbation Generation

### ISSUE THREAD

**Issue by Liangzhanbo** [yyliu01/PS-MT#44]
Hello, I would like to ask about the perturbation generation code `def get_r_adv_t(x, decoder1, decoder2, it=1, xi=1e-1, eps=10.0):`. The default value for the iteration steps it is set to 1, but a for loop format is used. Could you please clarify if setting a higher value for it would make sense in this case? Since the feature map is not updated during each iteration in the code, using a higher number of iterations seems to only explore the perturbation direction within the unit ball of the original feature map x_detached. 
Furthermore, the paper mentions "maximize" with respect to x_detached, but in the code, x.detach() is not updated during each iteration. Does the formula in the paper match the code implementation?

**Comment by yyliu01** [comment]
Hey @Liangzhanbo 

The implementation and setup is based on the original code [in here](https://github.com/lyakaap/VAT-pytorch/blob/75a3974887dc340eaf200f4715134eb882127b9c/vat.py#L25). In addition, the original paper has proved that 1-iteration is enough for finding the effective perturbation, 

![Image](https://github.com/user-attachments/assets/e2dae8d6-da3b-41e9-ac72-274d48527c4c)

The detached x is correct, because we want to find the most adversarial noise _d_ based on the fixed input feature map.

Cheers,
Yuyuan
